src/features/auto-resize-grid/js/auto-resize.js

```diff
@@ -19,25 +19,23 @@
       require: 'uiGrid',
       scope: false,
       link: function($scope, $elm, $attrs, uiGridCtrl) {
-
-        $scope.$watch(function() {
-          return $elm[0].clientWidth;
-        }, function(newVal, oldVal) {
-          if (newVal !== oldVal) {
-            uiGridCtrl.grid.gridWidth = newVal;
-            uiGridCtrl.grid.refresh();
+        $scope.$watchGroup([
+          function() {
+            return gridUtil.elementWidth($elm);
+          },
+          function() {
+            return gridUtil.elementHeight($elm);
+          }], function(newValues, oldValues, scope) {
+          if (newValues.toString() !== oldValues.toString()) {
```

Comment:
Why do you need to use the toString prior to comparing?

Reply:
I used toString because it's simples method how to compare two arrays in javascript. Method toString is supported in all browsers. And I can't directly compare two arrays because newValues === oldValues will always return false because it will compare if instance of array is same. It will not check if content is the same. But when I convert arrays to string, if the content is same, I will get two identical strings and is easy to compare two strings.

Reply:
Now when I think about it I changed it to angular.equals. It's better way in angular to compare arrays. Thanks for questioning.